translate/translator_yaml_test.go

```diff
@@ -0,0 +1,69 @@
+package translate_test
+
+import (
+	"testing"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/arr-ai/arrai/translate"
+	"github.com/stretchr/testify/require"
+)
+
+func AssertExpectedYAMLTranslation(t *testing.T, expected, rawYAML string) {
+	var data interface{}
+	require.NoError(t, yaml.Unmarshal([]byte(rawYAML), &data))
+	trans, err := translate.ToArrai(data)
+	require.NoError(t, err)
+	AssertExpectedTranslation(t, expected, trans)
+}
+
+func TestYAMLObjectToArrai(t *testing.T) {
+	t.Parallel()
+
+	// Empty
+	AssertExpectedYAMLTranslation(t, `{}`, `--- {}`)
+
+	// // different value types
+	AssertExpectedYAMLTranslation(t, `{"key": (v: 123)}                     `, `key: 123       `)
+	AssertExpectedYAMLTranslation(t, `{"key": (null: {})}                   `, `key:           `)
+	AssertExpectedYAMLTranslation(t, `{"key": (s: "val")}                   `, `key: val       `)
+	AssertExpectedYAMLTranslation(t, `{'key': (a: [(v: 1), (v: 2), (v: 3)])}`, `key: [ 1, 2, 3]`)
+	AssertExpectedYAMLTranslation(t, `{"key": {"foo": (s: "bar")}}          `, `key: {foo: bar}`)
+
+	// Multiple key-val pairs
+	AssertExpectedYAMLTranslation(t, `{"key": (s: "val"), "foo": (v: 123)}  `, `{"key":"val", "foo":123}`)
+}
+
+func TestYAMLArrayToArrai(t *testing.T) {
+	t.Parallel()
+
+	// Empty
+	AssertExpectedYAMLTranslation(t, `(a: [])`, `[]`)
+
+	// Different value types
+	AssertExpectedYAMLTranslation(t, `(a: [(v: 1)])                       `, `[1]            `)
+	AssertExpectedYAMLTranslation(t, `(a: [(null: {})])                   `, `[null]         `)
+	AssertExpectedYAMLTranslation(t, `(a: [(s: "hello")])                 `, `["hello"]      `)
+	AssertExpectedYAMLTranslation(t, `(a: [(a: [(v: 1), (v: 2), (v: 3)])])`, `[[1, 2, 3]]    `)
+	AssertExpectedYAMLTranslation(t, `(a: [{"foo": (s: "bar")}])          `, `[{"foo":"bar"}]`)
+
+	// Multiple values with different types
+	AssertExpectedYAMLTranslation(t, `(a: [(v: 1), (s: 'Hello'), (null: {})])`, `[1, "Hello", null]`)
+}
+
+func TestYAMLNullToNone(t *testing.T) {
+	t.Parallel()
+	AssertExpectedYAMLTranslation(t, `(null: {})`, `null`)
+}
+
+func TestYAMLStringToArrai(t *testing.T) {
+	t.Parallel()
+	AssertExpectedYAMLTranslation(t, `(s: {})           `, `""           `)
+	AssertExpectedYAMLTranslation(t, `(s: "Hello World")`, `"Hello World"`)
+}
+
+func TestYAMLNumericToArrai(t *testing.T) {
+	t.Parallel()
+	AssertExpectedYAMLTranslation(t, `(v: 123) `, `123 `)
```

Comment:
just realized, this is strange, numeric values shouldn't be converted to `(v: 123)`, it shouldn't be changed, why doesn't this fail?

Reply:
The result comes with `v:`, I'll dig into it and have a look.

Reply:
Found an issue in the translator, the switch case missed an int case